godal.h

```diff
@@ -25,6 +25,10 @@
 
 #if GDAL_VERSION_NUM < 3000000
 	#error "this code is only compatible with gdal version >= 3.0"
+#elif GDAL_VERSION_NUM < GDAL_COMPUTE_VERSION(3, 7, 0)
```

Comment:
I would prefer if this went into its own #if instead of this #elif, as the intent is different